File: 1615-maximal-network-rank/1615-maximal-network-rank.cpp
class Solution {
public:
    int maximalNetworkRank(int n, vector<vector<int>>& roads) {
        map<vector<int>,int>mp;
        for(auto it:roads){
            mp[it]++;
        }
        vector<int>indeg(n,0);
        for(int i=0;i<n;i++){
            for(auto it :roads){
                int f = it[0];
                int s = it[1];
                if(i==f || i==s){
                    indeg[i]++;
                }
            }
        }
        int res=-1;
        for(int i=0;i<n-1;i++){
            for(int j=i+1;j<n;j++){
                vector<int> check = {i,j};
                if(mp.find(check)!=mp.end() || mp.find({j,i})!=mp.end()){
                    res = max(res,indeg[i]+indeg[j]-1);
                }else{
                    res = max(res,indeg[i]+indeg[j]);
                }
            }
        }
        return res;
    }
};

File: 1615-maximal-network-rank/1615-maximal-network-rank.py
class Solution:
    def maximalNetworkRank(self, n: int, roads: List[List[int]]) -> int:
        mp = {} 
        for it in roads:
            mp[(it[0], it[1])] = 1
        indeg = [0] * n
        for i in range(0,n):
            for it in roads:
                f = it[0]
                s = it[1]
                if i==f or i==s:
                    indeg[i]+=1
                
        res=-1 
        for i in range(0,n-1):
            for j in range(i+1,n):
                if (i,j) in mp or (j,i) in mp:
                    res = max(res,indeg[i]+indeg[j]-1)
                else:
                    res = max(res,indeg[i]+indeg[j])
        return res